src/utils/encryption.ts

```diff
@@ -21,28 +28,28 @@ export class AES {
    * @param key - key to encrypt with
    * @returns ciphertext bundle
    */
-  static encryptGCM(plaintext: BytesData[], key: BytesData): Ciphertext {
+  static encryptGCM(plaintext: string[], key: string | Uint8Array): Ciphertext {
     // If types are strings, convert to bytes array
-    const plaintextFormatted = plaintext.map((block) => new Uint8Array(arrayify(block)));
-    const keyFormatted = new Uint8Array(arrayify(key));
+    const keyFormatted = typeof key === 'string' ? fastHexToBytes(key) : key;
     if (keyFormatted.byteLength !== 32) {
       throw new Error(
         `Invalid key length. Expected 32 bytes. Received ${keyFormatted.byteLength} bytes.`,
       );
     }
 
     const iv = AES.getRandomIV();
-    const ivFormatted = new Uint8Array(arrayify(iv));
+    const ivFormatted = fastHexToBytes(iv);
 
     // Initialize cipher
     const cipher = createCipheriv('aes-256-gcm', keyFormatted, ivFormatted, {
       authTagLength: 16,
     });
 
     // Loop through data blocks and encrypt
-    const data = plaintextFormatted
-      .map((block) => cipher.update(block))
-      .map((block) => block.toString('hex'));
```

Comment:
These maps were particularly bad, because there were 350k items, leading to on-the-fly creation of temporary 350k-large arrays.